force-app/main/default/classes/StringGeneratorTest.cls

```diff
@@ -0,0 +1,112 @@
+
+        String outputString;
+
+        Test.startTest();
+            outputString = StringGenerator.randomString( inputList );    
+        Test.stopTest();
+
+        System.assert( inputList.contains( outputString ), ' randomString - when passed a stringList, returns a value from the given stringList');
+    }
+
+    @isTest static void uniquelyRandomString_whenStringListIsNull_throwsException() //NOPMD - unit test naming convention followed 
+    {
+        Exception expectedError;
+
+        Test.startTest();
+            try 
+            {
+                StringGenerator.uniquelyRandomString( null );    
+            } 
+            catch ( Contract.RequiresException error ) 
+            {
+                expectedError = error;
+            }
+        Test.stopTest();
+
+        System.assertNotEquals ( null, expectedError, 'uniquelyRandomString - when passed a null stringList, throws a Contract.Requires Exception' ); 
+    }
+
+    @isTest static void uniquelyRandomString_whenStringListIsEmpty_returnsEmptyList() //NOPMD - unit test naming convention followed
+    {
+        List<String> inputList = new List<String>(); 
+
+        String outputString;
+
+        Test.startTest();
+            outputString = StringGenerator.uniquelyRandomString( inputList );    
+        Test.stopTest();
+
+        System.assertEquals( '', outputString, ' uniquelyRandomString - when passed an empty stringList, returns an empty string' );
+    }
+
+    @isTest static void uniquelyRandomString_whenGivenStringListWithValues_returnsRandomStringFromGivenStringList() //NOPMD - unit test naming convention followed 
+    {
+        List<String> inputList = new List<String>{ 'Value1', 'Value2' }; 
+
+        String outputString;
+
+        Test.startTest();
+            outputString = StringGenerator.uniquelyRandomString( inputList );    
+        Test.stopTest();
+
+        System.assert( inputList.contains( outputString ), ' uniquelyRandomString - when passed a stringList, returns a value from the given stringList');
+    }
+
+    @isTest static void uniquelyRandomString_whenAllStringsFromStringListAreServed_emptiesServedListToRestart() // NOPMD - unit test naming convention followed
+    {
+        List<String> inputList = new List<String>{ 'Value1', 'Value2' };
+
+        String outputString1,
+
+               outputString2,
+
+               outputString3;
+        
+        Test.startTest();
+            outputString1 = StringGenerator.uniquelyRandomString( inputList );
+
+            outputString2 = StringGenerator.uniquelyRandomString( inputList );
+            
+            outputString3 = StringGenerator.uniquelyRandomString( inputList );
+        Test.stopTest();
+
+        System.assert( ( inputList.contains( outputString1 ) ) && ( inputList.contains( outputString2 ) ) && ( inputList.contains( outputString3 ) ) , ' uniquelyRandomString - when all strings are served, restarts from beginning');
+
```

Comment:
one more test could be to check all 3 returned strings are not equal to each other, so that the uniqueness would have been verified by the unit test

Reply:
You are right. Will add it right away